src/DB_Command.php

```diff
@@ -1632,4 +1679,21 @@ private static function get_mysql_args( $assoc_args ) {
 
 		return $mysql_args;
 	}
+
+	/**
+	 * Writes out the `--no-defaults` flag for MySQL commands unless the --defaults flag is specified for the WP_CLI command.
+	 *
+	 * @param array $assoc_args Associative args array.
+	 * @return string Either the '--no-defaults' flag for use in the command or an empty string.
+	 */
+	private static function get_no_defaults( &$assoc_args ) {
```

Comment:
I think this method name doesn't read well and might be confusing (i.e., we're not actually fetching or not fetching any "defaults").

I'd suggest using something like this instead:
```suggestion
	private static function get_defaults_flag_string( &$assoc_args ) {
```

Reply:
Also, please use a regular protected method (not a private static one), and access it via `$this->`.

- I don't know why the helper methods in here are static, but that is not something to replicate, as static state is always a liability and should only be used if actually necessary. I'll fix the others in a subsequent PR.

- Using `private` within a class that is not `final` is a latent bug. Anyone extending this class and trying to override one of its public methods will not have access to the private methods it depends on. If they then try to override the private methods, they'll end up with two different versions of the method.

- Using `self` within a class that is not `final` is a latent bug as well. If someone overrides the class, any reference to `self` in the existing code will not take the extending class into account. `self` should be replaced with `static` in that case, to make use of late-static-binding.